packages/calypso-e2e/src/lib/components/sidebar-component.ts

```diff
@@ -35,22 +39,83 @@ export class SidebarComponent extends BaseContainer {
+	 * This method supports any of the following use cases:
+	 *   - heading only
+	 *   - subheading only
+	 *   - heading and subheading
+	 *
+	 * Heading is defined as the top-level menu item that is permanently visible on the sidebar, unless outside
+	 * of the viewport.
+	 *
+	 * Subheading is defined as the child-level menu item that is exposed only on hover or by toggling open the listing by clicking on the parent menu item.
+	 *
+	 * Note, in the current Nav Unification paradigm, clicking on certain combinations of sidebar menu items will trigger
+	 * navigation away to an entirely new page (eg. wp-admin). Attempting to reuse the SidebarComponent object
+	 * under this condition will throw an exception from the Playwright engine.
+	 *
+	 * @param {{[key: string]: string}} param0 Named object parameter.
+	 * @param {string} param0.heading Plaintext representation of the top level heading.
+	 * @param {string} param0.subheading Plaintext representation of the child level heading.
 	 * @returns {Promise<void>} No return value.
 	 */
-	async clickMenuItem( name: string ): Promise< void > {
-		await this.page.click( `text=${ toTitleCase( name ) }` );
+	async gotoMenu( {
+		heading,
+		subheading,
+	}: {
+		heading: string;
+		subheading?: string;
+	} ): Promise< void > {
+		if ( heading ) {
+			heading = toTitleCase( heading.trim() );
+			await this._click( { selector: `${ selectors.menuText } >> text=${ heading }` } );
+		}
+
+		if ( subheading ) {
+			subheading = toTitleCase( subheading.trim() );
+			await this.sidebar.waitForSelector( selectors.expandedMenu );
+			await this._click( { selector: `span:has-text("${ subheading }")`, force: true } );
+		}
+	}
+
+	/**
+	 * Performs the underlying click action on a sidebar menu item.
+	 *
+	 * This method ensures the sidebar is in a stable, consistent state prior to executing its actions.
+	 *
+	 * In some cases, due to the way WPCOM sidebar is implemented, Playwright will struggle to scroll the menu item
+	 * into the viewport in order to perform the click action. This is a particular issue with child-level menu
+	 * items (eg. Settings > Reading). Set the `force` parameter to true to force a click action to occur regardless
+	 * of the visibility state of the target element.
+	 *
+	 * @param {{[key: string]: string|boolean}} param0 Named object parameter.
+	 * @param {string} param0.selector Any selector supported by Playwright.
+	 * @param {boolean} [param0.force] Whether to force a click action. Defaults to false.
+	 * @returns {Promise<void>} No return value.
+	 */
+	async _click( {
+		selector,
+		force = false,
+	}: {
+		selector: string;
+		force?: boolean;
+	} ): Promise< void > {
+		await this.sidebar.waitForElementState( 'stable' );
+
+		const element = await this.sidebar.waitForSelector( selector );
+		await element.scrollIntoViewIfNeeded();
+
+		if ( ! force ) {
+			await element.click();
+		} else {
+			await element.dispatchEvent( 'click' );
```

Comment:
I've touched on this in the JSDoc for this method, but for some more background - 

It appears the scrolling behavior of WPCOM dashboard causes issues with Playwright.
Take for example `Settings > Reading`. On a 1400x1000 screen, make the `Reading` menu item is outside of the viewport, but attached to the DOM and 'visible' from Playwright's perspective.

With a normal `element.click()`, it would normally perform these checks:

- ensure element is actionable
- scroll element into view
- simulate a mouse click
- wait for navigation

Playwright appears to falter at the second check. 
Detailed browser API logs show this clearly:
```
pw:api   selector resolved to visible <span data-e2e-sidebar="Reading" class="sidebar__menu-…>Reading</span> +12ms
  pw:api <= elementHandle.waitForSelector succeeded +3ms
  pw:api => elementHandle.scrollIntoViewIfNeeded started +0ms
  pw:api   waiting for element to be visible and stable +1ms
  pw:api   element is visible and stable +47ms
  pw:api <= elementHandle.scrollIntoViewIfNeeded succeeded +1ms
  pw:api => elementHandle.click started +1ms
  pw:api attempting click action +0ms
  pw:api   waiting for element to be visible, enabled and stable +0ms
  pw:api   element is visible, enabled and stable +47ms
  pw:api   scrolling into view if needed +0ms
  pw:api   done scrolling +1ms
  pw:api   element is outside of the viewport +1ms
  pw:api retrying click action, attempt #1 +0ms
```

This is not an issue if the `dispatchEvent('click')` alternative approach is used, since it is able to force the event to occur without waiting for the checks. To ensure that navigation take place, an additional `waitForNavigation()` is placed.

Reply:
An alternative approach is to send a few `Keyboard.PageDown` events to physically scroll the page. 

This approach would eliminate the use of events to drive the interaction, but introduce ambiguity. How much scrolling is enough? What if viewport is so small the scrolling needs to occur varying number of times depending on the menu item being clicked?

I personally prefer to drive the interaction using the current method which while not simulating user interaction 100% it offers more certainty.

Reply:
If we use a custom scroll function using JS, not using the playwright command, does it work? We could re-use the scrolling from the current `scrollIntoView` method

Reply:
> If we use a custom scroll function using JS, not using the playwright command, does it work? We could re-use the scrolling from the current `scrollIntoView` method

@bsessions85 You know, that's a good point - I will try adapting `scrollIntoView` to this method.

Reply:
I've tried a couple of approaches and it did not seem to work:

```
			await this.page.$eval( selector, el => el.scrollIntoView({block: 'center', inline: 'center'}));
			await element.evaluate( el => el.scrollIntoView( {block: "start", inline: "center" }));
```

I don't think this is due to any A/B tests or Nav Unification. I suspect this issue has something to do with how both the sidebar and the main content has scroll bars, but I can't confirm.

I have not made any changes as a result.

Reply:
`scrollIntoView` doesn't work most probably because of the programmatic stickiness control of the sidenav position (don't know what's the proper name for this behavior, so here's a gif 😅):

![Jun-15-2021 15-40-24](https://user-images.githubusercontent.com/1451471/122062986-2d40d200-cdf0-11eb-861b-119da9638def.gif).

`window.scrollTo( x, y )` should still work though, so we only need to know how far we want to scroll it. Something like this should work:

```js
const element = await this.page.waitForSelector( selector );
const scrollToY = await element.boundingBox().y;

await this.page.evaluate( () => {
	window.scrollTo( 0, scrollToY );
} );
```

Reply:
There is a native `elementHandle.boundingBox()` method that will return `(x, y)` and `(width, height)` of the element's bounding box, so that should work.

With that said though, I am not sure if writing code to get the bounding box then scrolling would be an improvement over `dispatchEvent`, especially since the current implementation results in the expected functionality.

Personally, I do not think the benefits outweigh the complexity and time cost of rewriting this portion of the code. What do you think?

Reply:
> There is a native elementHandle.boundingBox() method that will return (x, y) and (width, height) of the element's bounding box, so that should work.

I used the original `getBoundingClientRect` on purpose here as it has the `bottom` property that the PW's `boundingBox` doesn't. Just one thing less to compute 😄 

> With that said though, I am not sure if writing code to get the bounding box then scrolling would be an improvement over dispatchEvent, especially since the current implementation results in the expected functionality.
>
> Personally, I do not think the benefits outweigh the complexity and time cost of rewriting this portion of the code. What do you think?

I think we can solve this by answering 2 questions (which is what I usually do in these kinds of situations):
1. Is it the best we can do to simulate the actual user interacting with the site, and
2. (How) is this going to affect debugging.

RE q1, the scroll&click would be our choice because this is how a human operates the site. The element needs to be in the viewport in order to be interacted with.
RE q2, the forced click could make it harder to debug because we wouldn't see the action on a recording of a failed test, as it would be happening off the viewport. Another point for the scroll&click implementation.

Does that make sense?

Reply:
Thanks for explaining your thoughts. 

I agree, and will incorporate your suggested code :)